extensions/binding/org.eclipse.smarthome.binding.mqtt.generic/src/main/java/org/eclipse/smarthome/binding/mqtt/generic/internal/values/RollershutterValue.java

```diff
@@ -55,29 +55,33 @@ public RollershutterValue(@Nullable String upString, @Nullable String downString
     @Override
     public void update(Command command) throws IllegalArgumentException {
         if (command instanceof UpDownType) {
-            PercentType newState = ((UpDownType) command).as(PercentType.class);
-            if (newState == null) { // Never happens
-                return;
-            }
-            state = newState;
+            throw new IllegalStateException("Cannot call update() with UpDownType");
```

Comment:
Why? UpDownType implements State, so updates are valid (and correspond to 0 and 100).

Reply:
So they are not meant to be move commands like "move up", "move down" but more like states meaning "completely up" and "completely down"?
A user wrote me a pm saying that stop doesn't work for him (which is absolutely correct) and that up/down sends his rollershutters up/down (which is correct so far) but also updates the state to 0% and 100% which he was not expecting.

Reply:
Yes, the commands UP/DOWN mean that it should move to UP (=0) or DOWN (=100) position.

> also updates the state to 0% and 100% 

The binding itself should indeed not update the state, but by default the "autoupdate" feature of the framework does it - if it sees an UP event, it will do a state auto-update to 0%.

Note that since https://github.com/eclipse/smarthome/pull/5011, the binding can define an auto-update policy, so you can actually configure channels in a way that the framework suppresses the auto-update behavior. This is probably what should usually be done for rollershutters as the state is reached much later than the command has been received, so the state update should better come from the physical device.

Reply:
@davidgraeff What's the status here now? Are you waiting for anything from my side?

Reply:
I have changed the code already and was waiting for another review. Have no time for the rebase right now though.

Reply:
I cannot see any code change, this is why I was asking. The line that I commented still throws an `IllegalStateException`...

Reply:
Ah ok, must have overlooked that line. When I do the rebase, I'll look into it again :)

Reply:
Any news here?

Reply:
I have added some changes